File: easy/two-number-sum/two_number_sum.py
def twoNumberSum(array, targetSum):
    # Write your code here.
    solutions = {
		elem: targetSum - elem
		for elem in array
	}
	
    for i, elem in enumerate(array):
	    if solutions[elem] in array and solutions[elem] in array[i+1:]:
		    return [elem, solutions[elem]]
	
    return []

File: easy/two-number-sum/two_number_sum.cpp
using namespace std;

bool isValidSubsequence(vector<int> array, vector<int> sequence) {
// Write your code here.
    int seq_index = 0;
    for (int i = 0; i < array.size(); i++) {
        if (array[i] == sequence[seq_index]) {
            seq_index += 1;
        }
        if (seq_index == sequence.size()) {
            return true;
        }
    }
return false;
}